cpp/benchmarks/common/generate_input.cu

```diff
@@ -604,38 +688,16 @@ std::unique_ptr<cudf::table> create_random_table(std::vector<cudf::type_id> cons
                                                  data_profile const& profile,
                                                  unsigned seed)
 {
-  cudf::size_type const num_cols = dtype_ids.size();
-  auto seed_engine               = deterministic_engine(seed);
-
-  auto const processor_count            = std::thread::hardware_concurrency();
-  cudf::size_type const cols_per_thread = (num_cols + processor_count - 1) / processor_count;
-  cudf::size_type next_col              = 0;
-  std::vector<std::future<columns_vector>> col_futures;
-  random_value_fn<unsigned> seed_dist(
-    {distribution_id::UNIFORM, 0, std::numeric_limits<unsigned>::max()});
-  for (unsigned int i = 0; i < processor_count && next_col < num_cols; ++i) {
-    auto thread_engine         = deterministic_engine(seed_dist(seed_engine));
-    auto const thread_num_cols = std::min(num_cols - next_col, cols_per_thread);
-    std::vector<cudf::type_id> thread_types(dtype_ids.begin() + next_col,
-                                            dtype_ids.begin() + next_col + thread_num_cols);
-    col_futures.emplace_back(std::async(std::launch::async,
-                                        create_random_columns,
-                                        std::cref(profile),
-                                        std::move(thread_types),
-                                        std::move(thread_engine),
-                                        num_rows.count));
-    next_col += thread_num_cols;
-  }
+  auto seed_engine = deterministic_engine(seed);
+  thrust::uniform_int_distribution<unsigned> seed_dist;
 
   columns_vector output_columns;
-  for (auto& cf : col_futures) {
-    auto partial_table = cf.get();
-    output_columns.reserve(output_columns.size() + partial_table.size());
-    std::move(
-      std::begin(partial_table), std::end(partial_table), std::back_inserter(output_columns));
-    partial_table.clear();
-  }
-
+  std::transform(
+    dtype_ids.begin(), dtype_ids.end(), std::back_inserter(output_columns), [&](auto tid) mutable {
```

Comment:
Why does this need to be mutable?

Reply:
Yes. `seed_engine` would be modified inside the lambda.